applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_analysis_stage.py

```diff
@@ -140,42 +159,37 @@ def _ExecuteInitialize(self):
         # Initialize IO
         self._InitializeIO()
 
-        ## Creation of the Kratos model (build sub_model_parts or submeshes)
-        self.structure_model = KratosMultiphysics.Model()
-        self.structure_model.AddModelPart(self.main_model_part)
+        ## Add the Modelpart to the Model if it is not already there
+        if not self.using_external_model_part:
+            self.model.AddModelPart(self.main_model_part)
 
         ## Print model_part and properties
         if self.is_printing_rank and self.echo_level > 1:
             KratosMultiphysics.Logger.PrintInfo("ModelPart", self.main_model_part)
             for properties in self.main_model_part.Properties:
                 KratosMultiphysics.Logger.PrintInfo("Property " + str(properties.Id), properties)
 
-        ## Processes construction
-        import process_factory
-        self.list_of_processes = process_factory.KratosProcessFactory(self.structure_model).ConstructListOfProcesses(self.ProjectParameters["constraints_process_list"])
-        self.list_of_processes += process_factory.KratosProcessFactory(self.structure_model).ConstructListOfProcesses(self.ProjectParameters["loads_process_list"])
-        if (self.ProjectParameters.Has("list_other_processes") == True):
-            self.list_of_processes += process_factory.KratosProcessFactory(self.structure_model).ConstructListOfProcesses(self.ProjectParameters["list_other_processes"])
-        if (self.ProjectParameters.Has("json_output_process") == True):
-            self.list_of_processes += process_factory.KratosProcessFactory(self.structure_model).ConstructListOfProcesses(self.ProjectParameters["json_output_process"])
+    def _SetUpListOfProcesses(self):
+        from process_factory import KratosProcessFactory
+        factory = KratosProcessFactory(self.model)
+        self.list_of_processes = factory.ConstructListOfProcesses(self.project_parameters["constraints_process_list"])
```

Comment:
do we need to distinguish the origin? i mean "constraints_process_list", loads_process_list, etc. ?

Ifthis is not so, then it would be better to have everythign in a single list, since anyway wethen push it back to a single list of processes

Reply:
@RiccardoRossi they don't need to be in different ones (even though I find it good to destinguish what a process is being used for)

=> I don't know who and why did this
I talked to @jcotela and we think that if we do not destinguish the origin, then it could be put to the base class

Note that this BREAKS the current Interface, i.e. old cases will stop working!

If you are willing to pay this price we can do it

Reply:
I don't think this is my call... in the sense that i should not be the one to decide this. Thing is that if the goal is unifying than we should go that way...or decide not to, preferably for good technical reasons